src/tiler/sql/user_reservable.sql

```diff
@@ -12,14 +12,15 @@
     -- 'available'.  We use a GeoJSON layer in the UI for the user's
     -- reserved blockfaces, and when a blockface is deselected we want to
     -- be able to reselect it
-    WHEN reservation.user_id IS NOT DISTINCT FROM <%= user_id %> THEN 'available'
+    WHEN (block.is_available
+          AND reservation.user_id IS NOT DISTINCT FROM <%= user_id %>) THEN 'available'
```

Comment:
Looks like reservations are not being deleted after submitting a survey. Not sure if that is by design or not.